src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,37 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: String) {
```

Comment:
Now you'll want to change the return type of ```run()``` so it has a ```Result``` type. My suggestion is this for the signature:

```
fn run(devpath: String) -> Result<(), String> {
  ...
}
```

Once you do that, you'll get quite a few build errors, not surprisingly. There will be at least one complaint about the ```run()``` method. No surprise, there. You've changed the return type, but you haven't changed the return value to match the type. To fix it up so it builds (doesn't yet do the correct thing) you'll need to add a line, shown below.